native/src/main/java/io/ballerina/stdlib/xmldata/json/JsonCreator.java

```diff
@@ -0,0 +1,164 @@
+/*
+ *  Copyright (c) 2023, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package io.ballerina.stdlib.xmldata.json;
+
+import io.ballerina.runtime.api.TypeTags;
+import io.ballerina.runtime.api.creators.ValueCreator;
+import io.ballerina.runtime.api.types.ArrayType;
+import io.ballerina.runtime.api.types.RecordType;
+import io.ballerina.runtime.api.types.TupleType;
+import io.ballerina.runtime.api.types.Type;
+import io.ballerina.runtime.api.utils.JsonUtils;
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.utils.TypeUtils;
+import io.ballerina.runtime.api.values.BArray;
+import io.ballerina.runtime.api.values.BError;
+import io.ballerina.runtime.api.values.BListInitialValueEntry;
+import io.ballerina.runtime.api.values.BMap;
+import io.ballerina.runtime.api.values.BString;
+
+import java.util.HashMap;
+import java.util.Iterator;
+
+import static io.ballerina.stdlib.xmldata.json.JsonParser.StateMachine.FIRST_ARRAY_ELEMENT_READY_STATE;
+import static io.ballerina.stdlib.xmldata.json.JsonParser.StateMachine.FIRST_FIELD_READY_STATE;
+
+/**
+ * Create objects for partially parsed json.
+ *
+ * @since 3.0.0
+ */
+public class JsonCreator {
+    static BArray finalizeArray(JsonParser.StateMachine sm, Type arrType, BArray currArr)
```

Comment:
Formating issue.

Reply:
I'm not sure what the issue here is, can you point it out?